tests/win_named_pipe.rs

```diff
@@ -0,0 +1,267 @@
+    assert!(events
+        .iter()
+        .any(|e| { e.token() == Token(1) && e.is_writable() }));
+}
+
+#[test]
+fn write_then_read() {
+    let (mut server, mut client) = pipe();
+    let mut poll = t!(Poll::new());
+    t!(poll.registry().register(
+        &mut server,
+        Token(0),
+        Interest::READABLE | Interest::WRITABLE,
+    ));
+    t!(poll.registry().register(
+        &mut client,
+        Token(1),
+        Interest::READABLE | Interest::WRITABLE,
+    ));
+
+    let mut events = Events::with_capacity(128);
+    t!(poll.poll(&mut events, None));
+
+    assert_eq!(t!(client.write(b"1234")), 4);
+
+    loop {
+        t!(poll.poll(&mut events, None));
+        let events = events.iter().collect::<Vec<_>>();
+        if let Some(event) = events.iter().find(|e| e.token() == Token(0)) {
+            if event.is_readable() {
+                break;
+            }
+        }
+    }
+
+    let mut buf = [0; 10];
+    assert_eq!(t!(server.read(&mut buf)), 4);
+    assert_eq!(&buf[..4], b"1234");
+}
+
+#[test]
+fn connect_before_client() {
+    let (mut server, name) = server();
+    let mut poll = t!(Poll::new());
+    t!(poll.registry().register(
+        &mut server,
+        Token(0),
+        Interest::READABLE | Interest::WRITABLE,
+    ));
+
+    let mut events = Events::with_capacity(128);
+    t!(poll.poll(&mut events, Some(Duration::new(0, 0))));
+    let e = events.iter().collect::<Vec<_>>();
+    assert_eq!(e.len(), 0);
+    assert_eq!(
+        server.connect().err().unwrap().kind(),
+        io::ErrorKind::WouldBlock
+    );
+
+    let mut client = client(&name);
+    t!(poll.registry().register(
+        &mut client,
+        Token(1),
+        Interest::READABLE | Interest::WRITABLE,
+    ));
+    loop {
+        t!(poll.poll(&mut events, None));
+        let e = events.iter().collect::<Vec<_>>();
+        if let Some(event) = e.iter().find(|e| e.token() == Token(0)) {
+            if event.is_writable() {
+                break;
+            }
+        }
+    }
+}
+
+#[test]
+fn connect_after_client() {
+    let (mut server, name) = server();
+    let mut poll = t!(Poll::new());
+    t!(poll.registry().register(
+        &mut server,
+        Token(0),
+        Interest::READABLE | Interest::WRITABLE,
+    ));
+
+    let mut events = Events::with_capacity(128);
+    t!(poll.poll(&mut events, Some(Duration::new(0, 0))));
+    let e = events.iter().collect::<Vec<_>>();
+    assert_eq!(e.len(), 0);
+
+    let mut client = client(&name);
+    t!(poll.registry().register(
+        &mut client,
+        Token(1),
+        Interest::READABLE | Interest::WRITABLE,
+    ));
+    t!(server.connect());
+    loop {
+        t!(poll.poll(&mut events, None));
+        let e = events.iter().collect::<Vec<_>>();
+        if let Some(event) = e.iter().find(|e| e.token() == Token(0)) {
+            if event.is_writable() {
+                break;
+            }
+        }
+    }
+}
+
+#[test]
+fn write_then_drop() {
```

Comment:
Re: not implementing `flush` (or a no-op implementation) above. I think running this test with the client in a different process that stops right after the write (line 194) would fail.